src/transformers.jl

```diff
@@ -79,6 +79,19 @@ function apply(A::AbstractArray, t::Transform; dims=:, inds=:, kwargs...)
     end
 end
 
+"""
+    apply!(A::AbstractArray, ::Transform; dims=:, kwargs...)
+
+Applies the [`Transform`](@ref) to each element of `A`, mutating the data.
+Optionally specify the `dims` to apply the [`Transform`](@ref) along certain dimensions.
+For example in a `Matrix`, `dims=1` applies to each column, while `dims=2` applies
+to each row.
+"""
+function apply!(A::AbstractArray, t::Transform; dims=:, kwargs...)
+    A[:] = apply(A, t; dims=dims, kwargs...)
```

Comment:
This was true before, but I noticed that `A[:] = ` doesn't work for `UnitRange`. I think that's to be expected (even on user end) because `UnitRange` is immutable.

Reply:
as in, if `A = 1:3` this kind of assignment won't work?

Reply:
Yep, simply because:

```
julia> x = 1:3
1:3

julia> x isa AbstractArray
true

julia> x[:] = 4:6
ERROR: setindex! not defined for UnitRange{Int64}
Stacktrace:
 [1] error(::String, ::Type{T} where T) at ./error.jl:42
 [2] error_if_canonical_setindex(::IndexLinear, ::UnitRange{Int64}, ::Int64) at ./abstractarray.jl:1161
 [3] setindex! at ./abstractarray.jl:1152 [inlined]
 [4] macro expansion at ./multidimensional.jl:802 [inlined]
 [5] macro expansion at ./cartesian.jl:64 [inlined]
 [6] macro expansion at ./multidimensional.jl:797 [inlined]
 [7] _unsafe_setindex! at ./multidimensional.jl:789 [inlined]
 [8] _setindex! at ./multidimensional.jl:785 [inlined]
 [9] setindex!(::UnitRange{Int64}, ::UnitRange{Int64}, ::Function) at ./abstractarray.jl:1153
 [10] top-level scope at REPL[39]:1
```